modules/qna/src/views/full/Components/QnA.tsx

```diff
@@ -0,0 +1,138 @@
+import { Button, Icon } from '@blueprintjs/core'
+import { confirmDialog, lang, MoreOptions } from 'botpress/shared'
+import cx from 'classnames'
+import _uniqueId from 'lodash/uniqueId'
+import React, { FC, Fragment, useEffect, useRef, useState } from 'react'
+
+import { MoreOptionsItems } from '../../../../../../src/bp/ui-shared/src/MoreOptions/typings'
+import style from '../style.scss'
+
+import TextareaList from './TextareaList'
+
+interface Props {
+  question: any
+  contentLang: string
+  updateQnA: (questions: any) => void
+  deleteQnA: () => void
+  toogleEnabledQnA: () => void
+}
+
+const QnA: FC<Props> = props => {
+  const [showOption, setShowOption] = useState(false)
+  const [collapsed, setCollapsed] = useState(!props.question.isNew)
+  const {
+    contentLang,
+    question: { data },
+    updateQnA
+  } = props
+  const questions = data.questions[contentLang]
+  const answers = data.answers[contentLang]
+  const focusedElement = useRef(`question-${questions.length - 1}`)
+
+  // Generating unique keys so we don't need to rerender all the list as soon as we add or delete one element
+  const questionKeys = useRef([])
+  const answerKeys = useRef([])
+
+  useEffect(() => {
+    questionKeys.current = [...Array(questions.length)].map(x => _uniqueId())
+    answerKeys.current = [...Array(answers.length)].map(x => _uniqueId())
+  }, [])
+
+  const onDelete = async () => {
+    if (
+      await confirmDialog(lang.tr('module.qna.form.confirmDeleteQuestion'), {
+        acceptLabel: lang.tr('delete'),
+        declineLabel: lang.tr('cancel')
+      })
+    ) {
+      props.deleteQnA()
+    }
+  }
+
+  const moreOptionsItems: MoreOptionsItems[] = [
+    {
+      label: lang.tr(data.enabled ? 'module.qna.form.disableQuestion' : 'module.qna.form.enableQuestion'),
+      action: props.toogleEnabledQnA
+    },
+    {
+      label: lang.tr('module.qna.form.deleteQuestion'),
+      type: 'delete',
+      action: onDelete
+    }
+  ]
+
+  const getPlaceholder = (type, index) => {
```

Comment:
Try to type everything, especially in new development. Also, if you have a known list of types, list them instead of using only "string"